feeder/src/vote.ts

```diff
@@ -303,20 +303,21 @@ interface VoteArgs {
 
 export async function vote(args: VoteArgs): Promise<void> {
   const client = new LCDClient({
-    URL: args.lcdAddress,
-    chainID: args.chainID,
-    gasPrices: args.gasPrices,
+    URL: process.env['LCD_ADDRESS'] || args.lcdAddress,
+    chainID: process.env['CHAIN_ID'] || args.chainID,
+    gasPrices: process.env['GAS_PRICE'] || args.gasPrices,
   })
-  const rawKey: RawKey = await initKey(args.keyPath, args.password)
-  const valAddrs = args.validator || [rawKey.valAddress]
+  const rawKey: RawKey = await initKey(process.env['KEY_PATH'] || args.keyPath, process.env['PASSPHRASE'] || args.password)
+  const valAddrs = (process.env['VALIDATOR'] && process.env['VALIDATOR'].split(',')) || args.validator || [rawKey.valAddress]
   const voterAddr = rawKey.accAddress
-  const denoms = args.denoms.split(',').map((s) => s.toLowerCase())
+  const denoms = (process.env['DENOMS'] || args.denoms).split(',').map((s) => s.toLowerCase())
   const wallet = new Wallet(client, rawKey)
+  const source = (process.env['SOURCE'] && process.env['SOURCE'].split(',')) || args.source
 
   while (true) {
     const startTime = Date.now()
 
-    await processVote(client, wallet, args.source, valAddrs, voterAddr, denoms).catch((err) => {
+    await processVote(client, wallet, source, valAddrs, voterAddr, denoms).catch((err) => {
```

Comment:
I think `ArgumentParser()` would fail if `required` arguments are missing, but making them optional via `required: false` I think there should be a check deeper in the code, are those previously `required: true` arguments provided by the operator, in some way (via cmdline or env var). Command should fail and give the feedback that required information was not provided.